LinkageError with quarkus-agroal and camel-quarkus-activemq

### Describe the bug

When using `camel-quarkus-activemq` together with `quarkus-agroal` there is a LinkageError:

```
Caused by: java.lang.LinkageError: loader constraint violation in interface itable initialization for class io.agroal.pool.util.XAConnectionAdaptor: when selecting method 'javax.transaction.xa.XAResource javax.sql.XAConnection.getXAResource()' the class loader 'platform' for super interface javax.sql.XAConnection, and the class loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @2e008502 of the selected method's class, io.agroal.pool.util.XAConnectionAdaptor have different Class objects for the type javax.transaction.xa.XAResource used in the signature (javax.sql.XAConnection is in module java.sql of loader 'platform'; io.agroal.pool.util.XAConnectionAdaptor is in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @2e008502, parent loader 'app')
```

(wasn't sure if this should be reported here or to camel-quarkus so have also raised an issue there - https://github.com/apache/camel-quarkus/issues/3548)

### Expected behavior

No linkage error, dependencies work together without any extra configuration

### Actual behavior

In order to get this working you have to exclude the transitive `geronimo-jta_1.1_spec`  dependency included by `camel-quarkus-activemq`, e.g.:

```groovy
implementation ('org.apache.camel.quarkus:camel-quarkus-activemq') {
  exclude group: 'org.apache.geronimo.specs', module: 'geronimo-jta_1.1_spec'
}
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Darwin Josephs-MacBook-Pro.local 20.6.0 Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.2" 2019-01-15 OpenJDK Runtime Environment 18.9 (build 11.0.2+9) OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.3.3 ------------------------------------------------------------  Build time:   2021-12-22 12:37:54 UTC Revision:     6f556c80f945dc54b50e0be633da6c62dbe8dc71  Kotlin:       1.5.31 Groovy:       3.0.9 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          11.0.2 (Oracle Corporation 11.0.2+9) OS:           Mac OS X 10.16 x86_64

### Additional information

_No response_

--- a/extensions/narayana-jta/runtime/pom.xml
+++ b/extensions/narayana-jta/runtime/pom.xml
@@ -72,10 +72,15 @@
                         <provides>io.quarkus.transactions</provides>
                     </capabilities>
                     <excludedArtifacts>
+                        <!-- Reference JTA API du jour is jakarta.transaction:jakarta.transaction-api: all other variants need to be exiled -->
                         <excludedArtifact>org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec</excludedArtifact>
                         <excludedArtifact>org.jboss.spec.javax.transaction:jboss-transaction-api_1.3_spec</excludedArtifact>
                         <excludedArtifact>javax.transaction:jta</excludedArtifact>
                         <excludedArtifact>javax.transaction:javax.transaction-api</excludedArtifact>
+                        <excludedArtifact>org.apache.geronimo.specs:geronimo-jta_1.0.1B_spec</excludedArtifact>
+                        <excludedArtifact>org.apache.geronimo.specs:geronimo-jta_1.1_spec</excludedArtifact>
+                        <excludedArtifact>org.apache.geronimo.specs:geronimo-jta_1.2_spec</excludedArtifact>
+                        <excludedArtifact>org.glassfish.main.transaction:javax.transaction</excludedArtifact>
                     </excludedArtifacts>
                 </configuration>
             </plugin>

